core/src/com/unciv/logic/HexMath.kt

```diff
@@ -136,21 +136,32 @@ object HexMath {
         }
         for (i in 0 until distance) { // 8 to 10
             vectors += current.cpy()
-            vectors += origin.cpy().scl(2f).sub(current) // Get vector on other side of clock
+            if (worldWrap){
+                if ((distance != maxDistance))
+                    vectors += origin.cpy().scl(2f).sub(current) // Get vector on other side of clock
+            } else {
+                vectors += origin.cpy().scl(2f).sub(current) // Get vector on other side of clock
+            }
             current.add(1f, 1f)
         }
         for (i in 0 until distance) { // 10 to 12
             vectors += current.cpy()
-            vectors += origin.cpy().scl(2f).sub(current) // Get vector on other side of clock
+            if (worldWrap){
```

Comment:
If(!worldWrap)...
Else if()...

Reply:
That's a v0.4 feature
Reduced the size and indentation and propagated it to this version